t/15-shared-plugin-gru.t

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env perl
+# Copyright SUSE LLC
+# SPDX-License-Identifier: GPL-2.0-or-later
+
+use Test::Most;
+
+use Test::Warnings ':report_warnings';
+use Test::MockModule;
+use Test::MockObject;
+use Test::Mojo;
+use FindBin;
+use lib "$FindBin::Bin/lib", "$FindBin::Bin/../external/os-autoinst-common/lib";
+use OpenQA::Test::TimeLimit '5';
+use OpenQA::Test::Case;
+use OpenQA::Shared::Plugin::Gru;    # SUT
+
+ok my $gru = OpenQA::Shared::Plugin::Gru->new, 'can instantiate gru plugin';
+
+my %connect_info = (dsn => 'my_dsn', user => 'user', password => 'foo');
+my $storage = Test::MockObject->new->set_always(connect_info => [\%connect_info]);
+my $schema = Test::MockObject->new->set_always(storage => $storage)->set_true('search_path_for_tests');
+my %config = (misc_limits => {minion_job_max_age => 0});
+my $minion = Test::MockObject->new->set_true('on');
+my $log = Test::MockObject->new->set_true('level', 'info');
+my $under = Test::MockObject->new->set_true('to');
+my $routes = Test::MockObject->new->set_always(under => $under);
+my $app
+  = Test::MockObject->new->set_always(schema => $schema)->set_always(config => \%config)->set_always(minion => $minion)
+  ->set_always(log => $log)->set_always(routes => $routes)->set_true('plugin', 'helper');
+OpenQA::App->set_singleton($app);
+my $mock = Test::MockModule->new('OpenQA::Shared::Plugin::Gru');
+$mock->noop('_allow_unauthenticated_minion_stats');
+ok $gru->register($app, undef), 'can register gru plugin';
+
```

Comment:
This test should also check whether the Minion plugin and its PostgreSQL connection were setup correctly (with the expected credentials). You can also create small test objects with sub packages like in the worker tests if `Test::MockObject` is not sufficient for this.

Reply:
With the current design of OpenQA::Shared::Plugin::Gru the PostgreSQL connection object is created within the method `register` which is IMHO bad design and makes testing not directly possible. We would need to move the object creation outside to inject dependencies. Then we would be able to inject a testable dependency. But I can mock the Mojo::Pg module (not object) with Test::MockModule.

I think checking anything more than just the current unit is overkill but I will do it. Updated.